### DISCUSSION TITLE
Meet an db excpetion when testing backend without docker

### DISCUSSION THREAD

**Discussion by chenkuangqi** [fastapi/full-stack-fastapi-template#1130] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
def init_db(session: Session) -> None:
    # Tables should be created with Alembic migrations
    # But if you don't want to use migrations, create
    # the tables un-commenting the next lines
    from sqlmodel import SQLModel

    from app.core.engine import engine

    # This works because the models are already imported and registered from app.models
    SQLModel.metadata.create_all(engine)
```

### Description

Despite of the comments in the init_db saying that un-comment next lines to avoid using migrations, there must be a mistake in the code:
`from app.core.engine import engine` (Line 21). 

There is no module named app.core.engine.

Exception Content:
```
sqlalchemy.exc.OperationalError: (psycopg.OperationalError) connection failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?
```

I run initial_data.py and backend_pre_start.py separately. 

### Operating System

Linux

### Operating System Details

VM running Ubuntu

### Python Version

Python 3.11.0rc1

**Comment by auh83** [comment]
@chenkuangqi I want to use this template without Docker too with only the backend to make an api only project. I tried with virtuaenv, can you explain your process without Docker ?

**Reply by chenkuangqi** [reply]
I am wondering if you just wanna get access to the api docs page. If so, try follow the poetry part of the README file in /backend. After the installation of all dependencies, run `uvicorn app.main:app --reload` and visit `<localhost>:<port>\docs` in the explorer. 
(You may need to change some configurations in core.config)

**Comment by auh83** [comment]
> I am wondering if you just wanna get access to the api docs page. If so, try follow the poetry part of the README file in /backend. After the installation of all dependencies, run `uvicorn app.main:app --reload` and visit `<localhost>:<port>\docs` in the explorer. (You may need to change some configurations in core.config)

I installed the dependencies with poetry and when I lauch uvicor, I have an error 

PROJECT_NAME
  Field required [type=missing, input_value={}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.6/v/missing
POSTGRES_SERVER
  Field required [type=missing, input_value={}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.6/v/missing
POSTGRES_USER
  Field required [type=missing, input_value={}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.6/v/missing
POSTGRES_PASSWORD
  Field required [type=missing, input_value={}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.6/v/missing
FIRST_SUPERUSER
  Field required [type=missing, input_value={}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.6/v/missing
FIRST_SUPERUSER_PASSWORD
  Field required [type=missing, input_value={}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.6/v/missing

**Reply by chenkuangqi** [reply]
As I said earlier, these are the variable names in app.core.config. Try set them up manually and rerun uvicorn.

**Reply by auh83** [reply]
Where do I set the variables in the config.py ?

**Reply by auh83** [reply]
It works now, I changed the place of the .env file

**Reply by auh83** [reply]
Now I have an error :

sqlalchemy.exc.OperationalError: (psycopg.OperationalError) connection failed: could not receive data from server: Socket is not connected (0x00002749/10057)
could not send SSL negotiation packet: Socket is not connected (0x00002749/10057)

**Comment by chenkuangqi** [comment]
> It works now, I changed the place of the .env file

That is an unusual way.. 

Now we are facing similar situation, which may caused by the migration part of the code.
